pkg/controller/publishingstrategy/publishingstrategy_controller.go

```diff
@@ -116,6 +116,23 @@ func (r *ReconcilePublishingStrategy) Reconcile(request reconcile.Request) (reco
 		return reconcile.Result{}, err
 	}
 
+	// get a list of all ingress on cluster that has annotation owner cloud-ingress-operator
+	// and delete all non-default ingresses
+	for _, ingressController := range ingressControllerList.Items {
+		if ingressController.Name != "default" && ingressController.Annotations["Owner"] == "cloud-ingress-operator" {
+			log.Info(fmt.Sprintf("ingresscontroller to be deleted: %v", ingressController))
+			err := r.client.Delete(context.TODO(), &ingressController)
+			if err != nil {
+				log.Error(err, "failed to delete ingresscontroller")
+				return reconcile.Result{}, err
+			}
+		}
+	}
+
+	// wait 60 seconds for deletion to be completed
+	log.Info("waited 60 seconds for necessary ingresscontroller deletions")
+	time.Sleep(time.Duration(60) * time.Second)
```

Comment:
Is this a good candidate for `return reconcile.Result{RequeueAfter: time.Duration(60 * time.Second)}, nil` so the operator can handle other things?

Reply:
+1. However, we need to have the proper status set up. I will write up a card and come back to this once we have status and conditions for the publishingStrategy CR